### PR TITLE
plan: decorrelation enhancement.

### PR DESCRIPTION
**Pull request by winoros**
Some improvement for decorrelation.
If the apply plan meets the following things:
- its inner plan is `Selection->Agg`.
- the correlated columns only occur in selection's equal conditions.
- apply's join type is outer join.

We can add the correlated column as the group by field and pull up the equal conditions as the join's join key. So the apply can be removed.
Note that some aggregate functions' default value is not null. I add a projection and use a `ifnull` to do this. In the near future i'll use join's default value to achieve this.

Performance:
- For tpch-10g query2: `N/A` to `7s`.
- For tpch-10g query17: `N/A` to `3 min or so`.

PTAL @zz-jason @lamxTyler

### CODE REVIEW

**Comment by zz-jason** [comment]
@winoros explain test failed, we need to modify test cases in mysql-test

File: plan/decorrelate.go [hunk 1]
```diff
@@ -78,9 +79,48 @@ func (la *LogicalAggregation) canPullUp() bool {
...
 type decorrelateSolver struct{}
 
+func (s *decorrelateSolver) aggDefaultValueMap(agg *LogicalAggregation) map[int]*expression.Constant {
+	defaultValueMap := make(map[int]*expression.Constant)
+	for i, f := range agg.AggFuncs {
+		switch f.Name {
+		case ast.AggFuncBitOr, ast.AggFuncBitXor, ast.AggFuncCount:
+			defaultValueMap[i] = expression.Zero.Clone().(*expression.Constant)
+		case ast.AggFuncBitAnd:
+			defaultValueMap[i] = expression.One.Clone().(*expression.Constant)
```
[the remark this answers is not in the document]
Comment (winoros) [reply plan/decorrelate.go]: Yeah, it's wrong. I've fixed in local commit.

**Pull request merged**